Updating color docs (#284)
primer/design: content/foundations/color.mdx

  src="https://user-images.githubusercontent.com/6951037/146927448-cb518377-114a-4ab8-a37e-1eeffc8732c7.png"
/>

GitHub's UI offers a variety of different color modes. Every pattern in Primer is built to work across all color modes out of the box.

> When designing product interfaces in Figma, we recommend to design in light mode. This is best, because the primer components for Figma are only available in light mode.
> To preview your work in other modes, use the [Figma color mode plugin](https://www.figma.com/community/plugin/992128487074360945/Change-Color-Mode).

Reviewer: ```suggestion
When working on product interfaces in Figma, we recommend designing in light mode as our Primer Figma components are only available in light mode.
To preview your work in other modes, use the [Figma color mode plugin](https://www.figma.com/community/plugin/992128487074360945/Change-Color-Mode).
```

Author: Hey Katie, while I love this, it moves us from a grade 9 to a grade 14 reading experience. I am trying to keep it lower. Can we break it up into multiple sentences. This always makes it easier to read.

Reviewer 2: >it moves us from a grade 9 to a grade 14 reading experience
Hey @lukasoppermann, I'm a bit confused by the grade reading levels, is there somewhere that outlines what this means? 

@langermank's suggestion on line 23 feels straightforward to me - I don't know if using a conjunction makes the reading level that much more difficult (but I suppose this is subjective)

Author: So we are suggesting here: https://primer.style/design/foundations/content#1-keep-it-clear a 7th grade or below level. Which I find very hard to reach. However using https://hemingwayapp.com/ you can get at least to a somewhat simple level.

While we are all somewhat deep in the topic colors, tokens, etc. is pretty complex and new stuff for others. Considering not everybody is an english-native speaker, I do think we should consider making out docs as easy as possible.

Please don't get me wrong, I am happy to change & update the text. But I do feel we should strive for as easy as possible.

Reviewer 2: Ah, I don't think I remembered this reference or didn't know how literally to take it (what defines a 7th grade reading level). That's a fair point.

I don't feel that strongly about this - if you want to keep two sentences I would simplify them both, and small fix to remove the comma in the second sentence: 

```suggestion
When designing product interfaces in Figma, we recommend using light mode. This is best because the Primer Figma components are only available in light mode.
To preview your work in other modes, use the [Figma color mode plugin](https://www.figma.com/community/plugin/992128487074360945/Change-Color-Mode).
```

## How to use color for product UI

Primer delivers colors in the form of [design tokens](https://primer.style/primitives/colors). Design tokens are a layer of abstraction that allows better maintainability, consistancy and easy theming.

> For example you use `bg-default` for the background of the page and `fg-default` for the text color. If the user changes to a dark mode, we change the underlying color that those tokens reference, but the tokens stay the same. 

Reviewer: ```suggestion
For example, if a page uses `bg-default` for the background color and `fg-default` for the text color, when the color mode is changed from light to dark, the tokens remain the same while their values update to respect the color mode. 
```

Author: Same as above. 😢 Love the change, but 9→14

Author: This one is not so easy. I think we can't just put a fullstop in there. @langermank any idea of how to make it work with two sentences?

Reviewer 2: ```suggestion
For example use `bg-default` for the background of the page and `fg-default` for the text color. If the user changes to a dark mode, the underlying color that those tokens reference change, but the tokens stay the same. 
```

Author: Love it. 👍

<img
  width="960"
  alt="Same component shown in light mode and dark mode color palettes"
  src="https://user-images.githubusercontent.com/813754/187436947-875dbf46-1b26-42da-b96f-e7cbe315f0eb.png"
/>

#### Color design tokens are grouped based on their purpose:

- **Presentational:** To represent a color. For example, the color steps in the Primer scale are named by color and lightness, such as `scale.blue.5`. These design tokens don't support color modes.

>   As the system grows, it will provide more APIs that fit different color needs. Check the [Primer Primitives](https://github.com/primer/primitives/releases/) repository to follow along as we release new variables and systems.

Reviewer: Should this chunk remain at the end?

Author: What are you exactly referring to?

> As the system grows, it will provide more APIs that fit different color needs. Check the [Primer Primitives](https://github.com/primer/primitives/releases/) repository to follow along as we release new variables and systems.

Or Presentational?

Author: Nevermind, I got it now and totally agree.
